include/range/v3/action/reverse.hpp

```diff
@@ -0,0 +1,79 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_ACTION_REVERSE_HPP
+#define RANGES_V3_ACTION_REVERSE_HPP
+
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/range_traits.hpp>
+#include <range/v3/action/action.hpp>
+#include <range/v3/action/erase.hpp>
+#include <range/v3/algorithm/reverse.hpp>
+#include <range/v3/utility/iterator_concepts.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-actions
+        /// @{
+        namespace action
+        {
+            struct reverse_fn
+            {
+            private:
+                friend action_access;
+
+            public:
+                template<typename Rng, typename I = iterator_t<Rng>,
+                         CONCEPT_REQUIRES_(BidirectionalRange<Rng>() and Permutable<I>())>
```

Comment:
I wish we had a clang-format file... when I want to edit range-v3 I need to indent everything manually by inserting spaces, and working against the indentation mechanism of my editor (and the tab key, which for me just invokes clang-format on the whole file) :/ 

clang-format isn't perfect though :/